### ISSUE TITLE
Please STOP Providing Fingerprints

### ISSUE THREAD

**Issue by adrianmmiller** [chiteroman/PlayIntegrityFix#137]
Its real simple, please stop hard coding fingerprints in the module....

We do appreciate your module initially helping us achieve Integrity, but now its just harming the scene.

No more fingerprints, please......people will just have to learn and most of them, to stop being lazy

**Comment by LucaPisl** [comment]
I really hope this is just an out-of-season April Fool's joke or sarcasm I'm not picking up on.
This module is absolutely great as is (thank you chiteroman), and plug and play support is something that should absolutely also exist in the rooting scene. If you personally don't like it, that is fine, and you can use the [DEV build](https://github.com/chiteroman/PlayIntegrityFix/releases/tag/DEV-v1). 
That being said, please don't encourage the developer to do something that will harm the experience of **most** users _just_ for the sake of gatekeeping.

**Comment by R3velation** [comment]
The problem with providing fingerprints is that they are going to get banned quickly.  Not to mention that they are not an infinite resource.  We can only use fingerprints from certain devices from a certain time-frame.  Once these prints are gone, they're gone, and there's no way to make more of them.  This renders this entire effort unsustainable.  Not to mention that if this goes far enough, google could decide to cut their losses and ban the entire range of prints; mandating strong integrity or nothing.

Over the last couple years, we've been spoiled with being able to have our roots be hidden, have doing so be skid accessible, and not have to worry sustainability.  Life was good.  Play integrity has changed that.  Now we can no longer have our cakes and eat them too.  We must choose between two options.  Either we continue along this path and hope that google tires of banning before we run out of valid prints and that they don't go nuclear on us.  Alternatively, we give up on keeping skid accessibility, get our own props, and keep them secret as to not get banned.  As much as you might hate it, we don't have much of a choice on this.

TL;DR: By providing prints in the module, we are effectively reporting to google that this print should be banned and we don't have any way to make more.  If this continues, we will run out of valid prints or worse.

**Comment by xChAmeLIoNz** [comment]
While I think it's true this method will get more FPs banned faster, I also think there should be some sort of guide that helps those less tech-savvy (or just don't have enough time), and provide an easier way to find these working FPs and keeping them private.

Not everybody is "just" lazy.

**Comment by R3velation** [comment]
There is a guide on getting the props you need.  The hard part is finding a valid device to get them from.
https://xdaforums.com/t/module-play-integrity-fix-safetynet-fix.4607985/post-89189572

**Comment by adrianmmiller** [comment]
> I really hope this is just an out-of-season April Fool's joke or sarcasm I'm not picking up on. This module is absolutely great as is (thank you chiteroman), and plug and play support is something that should absolutely also exist in the rooting scene. If you personally don't like it, that is fine, and you can use the [DEV build](https://github.com/chiteroman/PlayIntegrityFix/releases/tag/DEV-v1). That being said, please don't encourage the developer to do something that will harm the experience of **most** users _just_ for the sake of gatekeeping.

No, im just a long time and infamous figure on the Android scene, who has been doing magisk modules and other things since before this dev came along....so no, i dont qualify for an opinion (as everyone does just quietly)

If you cant see the issue ive raised, and im by no means the 1st person to ask the dev to politely stop risking everyone's ability to continue to use their devices in the manner which we currently do, then you should perhaps educate yourself with how the fingerprints are used, and then misused when suddenly a thousand people use the same one, often within hours of a new module being released. Perhaps then you might understand why this module is now actively harming the community.

No ones "gatekeeping" (which btw is another nonsense millennial/gen z snowflake term) anything....in fact on TG i myself gave people a link and info on how to harvest fingerprints individually - oddly those posts were removed by the dev who has a habit of setting TG channel settings (and then changing them back) and were even reposted after that and still available (if only the lazy generations would use a search)

So no one, especially me is "gatekeeping"... we're actually trying to protect the community as a whole

Also "gatekeeping" and the nonsense language needs to die, i weep for humanity at the PC snowflake way this world is headed

I identify as a north sea otter named Nigel

**Comment by adrianmmiller** [comment]
> There is a guide on getting the props you need. The hard part is finding a valid device to get them from. https://xdaforums.com/t/module-play-integrity-fix-safetynet-fix.4607985/post-89189572

If thats the only guide being given then i see why the lazy wait to be spoonfed.

I shared, as mentioned, several times on the TG group (i retired from XDA and wont post there) a direct URL to a site which gave you all the info to make a PIF direct on the web page, no need to download a ROM, or even read a build.prop. Im far from the only one who did this. I know it was also suggested on XDA

But if you give the average person these days, conditioned to being handed everything, guess what they will do...sit and wait until someone throws them the answer....

**Comment by adrianmmiller** [comment]
So here, for yet another round (and the last time i'll post it) is what i suggested people do, cut & paste from my TG post, if youre still the type that wants to sit back and wait to be spoonfed, then youre actively harming the community, and it would be better off without you quite frankly. This is how i have my stockpile of working fingerprints, and i know (from thanks and DM's) that plenty of people have managed to exert minutes of effort to get at least one for themselves. This is NOT the only database of build.props values on the Internet, either. There is ZERO excuse if youre given direct help like this to still be sitting back holding out your hand to be spoonfed a fingerprint. 

=====================================================================
"Folks it s not hard to find prints to use...theres entire ROM firmware databases of them - not sure why people think the ONLY way is to download old ROM's and manually extract the details

for example, general tip:

- step 1:  go to https://desktop.firmware.mobi/  (this is just one firmware repository that displays fingerprints etc)
- step 2:  type in model name for an old device 
- step 3:  choose an old ROM for that device that has security patch or build date generally 2018 or earlier
- step 4:   everything on the resulting page you need to add to your pif is there - cut and paste into the equivalent field in the json file

if you want to keep using the default pif that comes with the module, this will hasten the end of the workaround. if you expect to be spoonfed a new pif each time this happens then you should expect to be disappointed. you should be prepared to do some work yourself, or stop using the module....

people shouldnt be asking the same thing every 10 minutes, if you read the XDA thread, most of the tips above have already been available for some time, and for some of us this stuff goes back years. please educate yourself and stop relying solely on devs and others...by doing so you'll contribute to keeping the workaround working longer...simples"

=====================================================================

**Comment by R3velation** [comment]
Well, now that's posted.  I see zero reason why there should be provided prints in the module at all.

As for the skids that are reading this issue and getting pissed.  Sink or swim.  Either read the above guide and learn to swim or sink in your own ignorance.

**Comment by adrianmmiller** [comment]
An additional bit of free advice to attach to my earlier post:

If you find a fingerprint you harvest from that site doesnt pass, first try reducing the FIRST_API_LEVEL from whatever the listed one is for the ROM build (example: 27, 28) to 24 and it may pass, if it still doesnt, then it will be one of the ones probably already burnt by chiteroman and this module

**Comment by Omegamoney** [comment]
I don't think google will just stop banning FPs if each one of us use our own props. In fact, if we all stop mass using the same fingerprint, google will have even more reason to simply require strong integrity, why bother banning FPs when each user is spoofing their own? At that point it would be smarter to just enforce strong verity...

I'm not saying the Dev should stop, and if he does stop, someone else will create a new plugin with the same purpose, I'm just saying that we're running from the inevitable, and stopping the mass usage of FPs won't change that.

Y'all seem overly mad over something none of us have any control over.

I'd like to also recommend some time off of the internet, all the unrelated lashing out on the comments above were unnecessary, childish and unrelated to the topic.

**Comment by palakharinkhede4** [comment]
> So here, for yet another round (and the last time i'll post it) is what i suggested people do, cut & paste from my TG post, if youre still the type that wants to sit back and wait to be spoonfed, then youre actively harming the community, and it would be better off without you quite frankly. This is how i have my stockpile of working fingerprints, and i know (from thanks and DM's) that plenty of people have managed to exert minutes of effort to get at least one for themselves. This is NOT the only database of build.props values on the Internet, either. There is ZERO excuse if youre given direct help like this to still be sitting back holding out your hand to be spoonfed a fingerprint.
> 
> ===================================================================== "Folks it s not hard to find prints to use...theres entire ROM firmware databases of them - not sure why people think the ONLY way is to download old ROM's and manually extract the details
> 
> for example, general tip:
> 
> * step 1:  go to https://desktop.firmware.mobi/  (this is just one firmware repository that displays fingerprints etc)
> * step 2:  type in model name for an old device
> * step 3:  choose an old ROM for that device that has security patch or build date generally 2018 or earlier
> * step 4:   everything on the resulting page you need to add to your pif is there - cut and paste into the equivalent field in the json file
> 
> if you want to keep using the default pif that comes with the module, this will hasten the end of the workaround. if you expect to be spoonfed a new pif each time this happens then you should expect to be disappointed. you should be prepared to do some work yourself, or stop using the module....
> 
> people shouldnt be asking the same thing every 10 minutes, if you read the XDA thread, most of the tips above have already been available for some time, and for some of us this stuff goes back years. please educate yourself and stop relying solely on devs and others...by doing so you'll contribute to keeping the workaround working longer...simples"
> 
> =====================================================================

I totally agree! The lazy ones should just lock the bootloaders and stay in peace. As for the less tech-savvy people, if they are willing to unlock bootloader or root their devices, they should also be willing to find a working fingerprint to spoof. Plus the process is fairly easy if one is ready to read and work a little. This spoonfeeding will eventually only harm the community. Hope Chiteroman gets this!

**Comment by xChAmeLIoNz** [comment]
For those who want to keep root and have a locked bootloader

[https://github.com/chenxiaolong/avbroot](https://github.com/chenxiaolong/avbroot)

**Comment by adrianmmiller** [comment]
> For those who want to keep root and have a locked bootloader
> 
> https://github.com/chenxiaolong/avbroot

that doesnt help with this issue....

**Comment by chiteroman** [comment]
No.
